src/planner/plan_expression.cpp

```diff
@@ -0,0 +1,143 @@
+#include "execution/expressions/comparison_expression.h"
+#include "execution/expressions/constant_value_expression.h"
+#include "execution/plans/abstract_plan.h"
+#include "fmt/format.h"
+#include "planner/planner.h"
+
+namespace bustub {
+
+const char *unnamed_column = "<unnamed>";
+
+auto Planner::PlanBinaryOp(const BoundBinaryOp &expr, const std::vector<const AbstractPlanNode *> &children)
+    -> std::unique_ptr<AbstractExpression> {
+  auto [_1, left] = PlanExpression(*expr.larg_, children);
+  auto [_2, right] = PlanExpression(*expr.rarg_, children);
+  const auto &op_name = expr.op_name_;
+
+  if (op_name == "=" || op_name == "==") {
+    return std::make_unique<ComparisonExpression>(SaveExpression(std::move(left)), SaveExpression(std::move(right)),
+                                                  ComparisonType::Equal);
+  }
+  if (op_name == "!=" || op_name == "<>") {
+    return std::make_unique<ComparisonExpression>(SaveExpression(std::move(left)), SaveExpression(std::move(right)),
+                                                  ComparisonType::NotEqual);
+  }
+  if (op_name == "<") {
+    return std::make_unique<ComparisonExpression>(SaveExpression(std::move(left)), SaveExpression(std::move(right)),
+                                                  ComparisonType::LessThan);
+  }
+  if (op_name == "<=") {
+    return std::make_unique<ComparisonExpression>(SaveExpression(std::move(left)), SaveExpression(std::move(right)),
+                                                  ComparisonType::LessThanOrEqual);
+  }
+  if (op_name == ">") {
+    return std::make_unique<ComparisonExpression>(SaveExpression(std::move(left)), SaveExpression(std::move(right)),
+                                                  ComparisonType::GreaterThan);
+  }
+  if (op_name == ">=") {
+    return std::make_unique<ComparisonExpression>(SaveExpression(std::move(left)), SaveExpression(std::move(right)),
+                                                  ComparisonType::GreaterThanOrEqual);
+  }
+
+  throw Exception(fmt::format("binary op {} not supported in planner yet", op_name));
+}
+
+auto Planner::PlanColumnRef(const BoundColumnRef &expr, const std::vector<const AbstractPlanNode *> &children)
+    -> std::tuple<std::string, std::unique_ptr<AbstractExpression>> {
+  if (children.empty()) {
+    throw Exception("column ref should have at least one child");
+  }
+
+  auto col_name = fmt::format("{}.{}", expr.table_, expr.col_);
+
+  if (children.size() == 1) {
+    // Projections, Filters, and other executors evaluating expressions with one single child will
+    // use this branch.
+    const auto &child = children[0];
+    auto schema = child->OutputSchema();
+    uint32_t col_idx = schema->GetColIdx(col_name);
+    auto col_type = schema->GetColumn(col_idx).GetType();
+    return std::make_tuple(col_name, std::make_unique<ColumnValueExpression>(0, col_idx, col_type));
+  }
+  if (children.size() == 2) {
+    /**
```

Comment:
```suggestion
    /*
```

Reply:
`/**` is for doxygen function blocks, general block comments can stick with `/*`